tests/matrix_multiply_tests/matrix_multiply.futil

```diff
@@ -0,0 +1,232 @@
+      j.write_en = 1'd1; 
+      let1[done] = j.done; 
+    }
+    group let2<"promotable"=1> { 
+      k.in = const1.out; 
+      k.write_en = 1'd1; 
+      let2[done] = k.done; 
+    }
+
+    // Accumulator Reset
+    group begin_acc {
+      acc.in = const1.out;
+      acc.write_en = 1'd1;
+      begin_acc[done] = acc.done;
+    }
+
+    group read_data {
+      slice0.in = i.out;
+      slice1.in = k.out;
+      slice2.in = j.out;
+      arr_1.addr0 = slice0.out; 
+      arr_1.addr1 = slice1.out;
+      arr_2.addr0 = slice1.out; 
+      arr_2.addr1 = slice2.out;
+
+      val_a.in = arr_1.read_data;
+      val_a.write_en = 1'd1;
+      val_b.in = arr_2.read_data;
+      val_b.write_en = 1'd1;
+      read_data[done] = val_b.done;
+    }
+
+    // group update_output {
+      
+    // }
+
+    // Matrix Multiplication Update
+    // group update_acc {
+    //   // mul_add.reg_done = 1'd1;
+    //   // mul_add.reg_out = acc.out;
+    //   slice0.in = i.out;
+    //   slice1.in = k.out;
+    //   arr_1.addr0 = slice0.out; 
+    //   arr_1.addr1 = slice1.out;
+    //   arr_2.addr0 = slice1.out; 
+    //   arr_2.addr1 = slice0.out;
+    //   // mul_add.a = arr_1.read_data;
+    //   // mul_add.b = arr_2.read_data;
+    //   acc.in = mul_add.reg_in;
+    //   acc.write_en = mul_add.reg_write_en;
+    //   update_acc[done] = acc.done;
+    // }
+
+    // Writing results
+    group write_result {
+      slice0.in = i.out;
+      slice1.in = j.out;
+      mem_output.addr0 = slice0.out;
+      mem_output.addr1 = slice1.out;
+      mem_output.write_data = acc.out;
+      mem_output.write_en = 1'd1;
+      write_result[done] = mem_output.done;
+    }
+
+    // Loop Increment
+    group updI<"promotable"=1> { 
+      add.left = i.out; 
+      add.right = const2.out; 
+      i.in = add.out; 
+      i.write_en = 1'd1; 
+      updI[done] = i.done; 
+    }
+    group updJ<"promotable"=1> { 
+      add.left = j.out; 
+      add.right = const2.out; 
+      j.in = add.out; 
+      j.write_en = 1'd1; 
+      updJ[done] = j.done; 
+    }
+    group updK<"promotable"=1> { 
+      add.left = k.out; 
+      add.right = const2.out; 
+      k.in = add.out; 
+      k.write_en = 1'd1; 
+      updK[done] = k.done; 
+    }
+    group resetI {
+      i.in = const1.out;
+      i.write_en = 1'd1;
+      resetI[done] = i.done;
+    }
+    group resetJ {
+      j.in = const1.out;
+      j.write_en = 1'd1;
+      resetJ[done] = j.done;
+    }
+    group resetK {
+      k.in = const1.out;
+      k.write_en = 1'd1;
+      resetK[done] = k.done;
+    }
+  }
+
+  control {
+    seq {
+      // i, j, k initialization
+      let0; 
+      let1; 
+      let2;
+      resetI;
+      while le0.out with cond0 {
+        seq {
+          resetJ;
+          while le1.out with cond1 {
+            seq {
+              resetK;
+              begin_acc; // Reset accumulator
+              while le2.out with cond2 {
+                seq { 
+                  // update_acc; 
```

Comment:
Please clean up the comments here! (or rearrange them to comment on the groups/invokes you have in your code)